src/layout/Checkboxes/CheckboxesContainerComponent.tsx

```diff
@@ -17,33 +17,22 @@ import type { PropsFromGenericComponent } from 'src/layout';
 
 export type ICheckboxContainerProps = PropsFromGenericComponent<'Checkboxes'>;
 
-const defaultSelectedOptions: string[] = [];
-
 export const CheckboxContainerComponent = ({ node, isValid, overrideDisplay }: ICheckboxContainerProps) => {
-  const { id, layout, readOnly, textResourceBindings, dataModelBindings, required, labelSettings, alertOnChange } =
-    node.item;
+  const { id, layout, readOnly, textResourceBindings, required, labelSettings, alertOnChange } = node.item;
   const { langAsString } = useLanguage();
 
-  const value = FD.usePickFreshString(dataModelBindings?.simpleBinding);
-  const setData = FD.useSetForBindings(dataModelBindings);
   const debounce = FD.useDebounceImmediately();
 
-  const selected = value && value.length > 0 ? value.split(',') : defaultSelectedOptions;
-  const { options: calculatedOptions, isFetching } = useGetOptions({
+  const {
+    options: calculatedOptions,
+    isFetching,
+    setData,
+    current,
+    currentStringy,
```

Comment:
I am a little confused by the name `currentStringy` :sweat_smile: Is it supposed to be the current option stringified? If so, I think a more verbose naming makes it easier to understand.

Reply:
Good point! I added some comments describing them. It's not always a string (for multiple selection components it's an array of strings), so that's why I named it 'stringy' not just 'string'.

Also, most of the point of this refactor was to make sure the source options, which are derived from the data model, actually was represented as string values even if the data model used other types (which it can now). I added an extensive unit test to make sure that's the case now for other configurations (such as static options in layout, along with options from API). Each one of them should now work correctly even if the data in them is a number or boolean. :partying_face: Thanks for nudging me to do that!